subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/DefaultBuildCacheController.java

```diff
@@ -100,7 +99,7 @@ public boolean isEmitDebugLogging() {
             try {
                 local.load(command.getKey(), unpack);
             } catch (Exception e) {
-                throw new GradleException("Build cache entry " + command.getKey().getHashCode() + " from local build cache is invalid", e);
+                throw BuildCacheLoadException.forLocalCache(command.getKey().getHashCode(), e);
```

Comment:
Yes, this is intentional. Since the original message was that `build cache is invalid`, which might not be always true. For example in the given example from reporter directory could not be deleted, but we reported that `build cache is invalid`.